mono/utils/mono-threads-coop.c

```diff
@@ -707,3 +707,17 @@ mono_threads_coop_end_global_suspend (void)
 	if (mono_threads_are_safepoints_enabled ())
 		mono_polling_required = 0;
 }
+
+void
+mono_threads_enter_no_safepoints_region (const char *func)
+{
+	MONO_REQ_GC_UNSAFE_MODE;
+	mono_threads_transition_begin_wont_safept (mono_thread_info_current (), func);
```

Comment:
So. In a separate PR, I suggest we have a global non-thread local `mono_thread_info_current_var`, pointer, const, always null. You pass that. At least in some macros, maybe not here.

A function needing the data, if it gets null, will call `mono_thread_info_current()`

Macros like HANDLE_FUNCTION_BEGIN that call `mono_thread_info_current()` anyway, will store in a local named `mono_thread_info_current_var`.

Therefore TLS accesses will be optimized down.

If/as perf sensitive functions are found, pass the local as a parameter.

Reply:
ok, I'll update to pass the thread info and reconcile with #11487 once it's in.